saiku-webapp/src/main/java/org/saiku/CCRSMondrianSchemaProcessor.java

```diff
@@ -0,0 +1,187 @@
+package org.saiku;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.text.Normalizer;
+import java.text.Normalizer.Form;
+import java.util.HashSet;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.sun.jersey.core.impl.provider.entity.Inflector;
+
+import mondrian.olap.Util.PropertyList;
+import mondrian.spi.DynamicSchemaProcessor;
+
+/**
+ * Generates the schema files based on reusable data and automatic configurations.
+ * First it does string based data injections based on TAGs. Then it customizes @@references@@.
+ * 
+ * @author Nadejda Mandrescu
+ */
+public class CCRSMondrianSchemaProcessor implements DynamicSchemaProcessor {
+    /*
+     * TODO
+     *  - schema caching in non-dev mode / check for changes?
+     */
+    
+    private static final Inflector INFLECTOR = Inflector.getInstance();
+    private static final Pattern YES_NO_DIM_PATTERN = Pattern.compile(
+            "<Dimension table=\"YesNoTable\" *name=[\"|']([^=]*)[\"|'] *caption=[\"|']([^=]*)[\"|'] */>");
+    private static final String YES_NO_DIM_TEMPLATE =
+            "<Dimension name='@@name@@' caption='@@caption@@' table='YesNoTable' key='Dimension Id'>\n" +
+                "<Attributes>\n" +
+                    "<Attribute name='Dimension Id' keyColumn='id' hasHierarchy='false' \n" +
+                                "levelType='Regular' datatype='Integer'/>\n" +
+                    "<Attribute name='@@name@@' caption='@@caption@@' keyColumn='answer' \n" +
+                                "approxRowCount='2' hierarchyHasAll='true' levelType='Regular' datatype='Boolean'/>\n" +
+                "</Attributes>\n" +
+            "</Dimension>";
+    private static final String CATEGORY_QUERY_TEMPLATE =
+            "<Query alias='@@table@@'>\n" +
+                "<ExpressionView>\n" +
+                    "<SQL dialect='generic'>\n" +
+                        "<![CDATA[SELECT ID, LABEL FROM CATEGORY WHERE DTYPE='@@table@@']]>\n" +
+                    "</SQL>\n" +
+                "</ExpressionView>\n" +
+             "</Query>\n";
+    private static final Pattern CATEGORY_DIM_PATTERN = Pattern.compile(
```

Comment:
To improve debugging, explain a bit what are you catching with longer regex patterns